pandas/tests/frame/test_api.py

```diff
@@ -224,6 +225,16 @@ def test_iterrows_iso8601(self):
             exp = s.loc[k]
             self._assert_series_equal(v, exp)
 
+    def test_iterrows_corner(self):
+        # gh-12222
+        df = DataFrame(
+            {'a': [datetime.datetime(2015, 1, 1)], 'b': [None], 'c': [None],
+             'd': [''], 'e': [[]], 'f': [set()], 'g': [{}]})
+        expected = df.iloc[0]
+        _, result = next(df.iterrows())
+        assert result.dtype == np.object_
```

Comment:
We already compare `dtype` in `assert_series_equal`, no?

Reply:
since the expected is not explicit, the dtype check is for added assurance.  so the dtype check could have easily been applied to the expected instead. and if the expected was a fixture it would error instead of fail, indicating a bad test.

Reply:
Hmm...perhaps we should construct the expected `Series` explicitly instead of relying on the result of `df.iloc[0]`.

Reply:
probably a good idea, i'll change it.